x-pack/plugins/cases/public/components/all_cases/utils/sanitize_filter_options.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FilterOptions } from '../../../../common/ui/types';
+import type { CaseStatuses, CaseSeverity } from '../../../../common/types/domain';
+
+const notAll = (option: string) => option !== 'all';
+
+/**
+ * In earlier versions, the options 'status' and 'severity' could have a value of 'all'.
+ * This function ensures such legacy values are removed from the URL parameters to maintain
+ * backwards compatibility.
+ */
+export const removeLegacyValuesFromOptions = ({
+  status: legacyStatus,
+  severity: legacySeverity,
+}: {
+  status: CaseStatuses[];
+  severity: CaseSeverity[];
+}) => {
+  return {
+    status: legacyStatus?.filter(notAll).filter(Boolean),
+    severity: legacySeverity?.filter(notAll).filter(Boolean),
+  };
+};
+
+export const removeNonStoreableFilters = (filterOptions: Partial<FilterOptions>) => {
+  const { assignees, category, owner, reporters, search, searchFields, tags, ...storableFilters } =
```

Comment:
I think is better if we are explicit on what we return. It is easier to control an allow list than a disallow list. In the we may add a new filter option and `storableFilters` will return that.

Reply:
Do we want to save the value of the custom fields filters chosen? I did it this way because I thought it's easier to filter out those field I know won't be stored in the local storage that the ones that should be stored once it's including the custom fields

Reply:
ok, returning explicitly as nothing else needed for now